src/storage/txn/scheduler.rs

```diff
@@ -422,6 +417,30 @@ fn process_write_impl(cid: u64,
             let pr = ProcessResult::Res;
             (pr, txn.modifies())
         }
+        Command::ResolveLock { ref ctx, start_ts, commit_ts, ref keys } => {
+            let mut txn = MvccTxn::new(snapshot, start_ts, None);
+            match commit_ts {
+                Some(ts) => {
+                    for k in keys {
+                        try!(txn.commit(&k, ts));
+                    }
+                }
+                None => {
+                    for k in keys {
+                        try!(txn.rollback(&k));
+                    }
+                }
+            }
+            let pr = ProcessResult::NextCommand {
```

Comment:
Will the next rolled back seek the locks from the beginning? If yes, we may skip many tombstone keys.